### ISSUE TITLE
Change Tab keybinding to accept Cursor ai suggestion only if there are not Intellisense suggestions

### ISSUE THREAD

**Issue by zanhk** [getcursor/cursor#2518]
I m looking forward to a solution for this as well as is pretty annoying.

Right now cursor keep messing on these cases:

- when you try to make an import and vscode know where the file is but cursor put a totally random path which doesn't even exist
- when you try to insert tailwind classes that are autocompleted from the extension but cursor decide to allucintate and invent random classess
- when you are filling any method where the ide alerady know the types and propose the following method but cursor decide that it does not like it
- Actually all the times there is any other autocomplete I never wished to have cursor autocomplete instead so It would be better to disable it at all if there is any other suggestion from a different source

To be clear I want to use cursor autocomplete with `tab` but only if there is not any other suggestion from the ide/plugin/snippets etc..

So in the keybindings would be like this

| Command |  Keybinding | When |
| --- | --- | --- |
|  Accept Cursor suggestion | Tab | !hasOtherSuggestions | 

Is there a way to archieve this right now? because like this is not very usable.

**Comment by nolanjj** [comment]
I agree!!!

**Comment by superjump22** [comment]
I agree!!!

**Comment by psychobunny** [comment]
@Moltenship thank you, works perfectly (had to also delete `acceptCursorTabSuggestion` in cursor keyboard shortcuts). Much less pulling out hair so far. 

Funny, you answered this five minutes before I started searching for the problem. Are you powered by AI too? 😆

**Comment by Moltenship** [comment]
@psychobunny Not really, just had a few minutes to fix something that was bothering me since I moved from VSCode + Supermaven   🙃 . Glad that it helped you

**Comment by zanhk** [comment]
@Moltenship Thanks

**Comment by zanhk** [comment]
https://github.com/user-attachments/assets/839d7280-06f2-4207-bb20-4e2a7085ba99


@Moltenship Actually is not working for me

**Comment by YinanZhaoXometry** [comment]
1.  Update to the latest version, this is my version:

![Image](https://github.com/user-attachments/assets/21a2f49a-ea64-4e8c-8a40-2d66dc81862c)

2. This shortcut setting works for me:
```json
  {
    "key": "tab",
    "command": "editor.action.acceptCursorTabSuggestion",
    "when": "cpp.shouldAcceptTab && !editorHoverFocused && !editorTabMovesFocus && !suggestWidgetVisible"
  },
  {
    "key": "tab",
    "command": "-editor.action.acceptCursorTabSuggestion"
  },
```

**Comment by HuanzhiMao** [comment]
Thank you @YinanZhaoXometry. This is very helpful :)

**Comment by caitoulin** [comment]
{
        "key": "tab",
        "command": "editor.action.cursorMateTabSuggestion",
        "when": "cpp.shouldAcceptTab && !editorHoverFocused && !editorTabMovesFocus && !suggestWidgetVisible"
      },
      {
        "key": "tab",
        "command": "-editor.action.acceptCursorTabSuggestion"
      }    

     const customTabBehaviorDisposable = vscode.commands.registerCommand('editor.action.cursorMateTabSuggestion', async (...args) => {
      await vscode.commands.executeCommand('editor.action.acceptCursorTabSuggestion', ...args);
    });

 "I want to intercept the acceptCursorTabSuggestion command, but why has the default tab functionality stopped working? It no longer provides indentation."

**Comment by caitoulin** [comment]
@YinanZhaoXometry
